psycopg/setup.cfg

```diff
@@ -55,7 +55,7 @@ packages = find:
 zip_safe = False
 install_requires =
     backports.zoneinfo >= 0.2.0; python_version < "3.9"
-    typing-extensions >= 4.4
+    typing-extensions >= 4.4; python_version < "3.13"
```

Comment:
Note: this change doesn't belong to this changeset. I'll push it separately to master.